As a first step, I only committed the test, and it failed as expected:
```
./src/ocean/text/convert/Formatter.d(549): Error: cannot modify `inout` expression `b[1]`
./src/ocean/text/convert/Formatter.d(279): Error: template instance `ocean.text.convert.Formatter.handle!(inout(char))` error instantiating
./src/ocean/text/convert/Formatter.d(137):        instantiated from here: `sformat!(inout(char))`
./src/ocean/text/convert/Formatter_test.d(575):        instantiated from here: `format!(inout(char))`
make: *** [/home/runner/work/ocean/ocean/build/prod/tmp/allunittests] Error 1
```

--- a/src/ocean/text/convert/Formatter.d
+++ b/src/ocean/text/convert/Formatter.d
@@ -545,8 +545,7 @@ private void handle (T) (T v, FormatInfo f, scope FormatterSink sf, scope ElemSi
                     || is(typeof((&v)[0 .. 1]) : const(wchar)[])
                     || is(typeof((&v)[0 .. 1]) : const(dchar)[]))
     {
-        Unqual!(T)[3] b = "'_'";
-        b[1] = v;
+        T[3] b = [ '\'', v, '\'' ];
         if (f.flags & Flags.Nested)
             UTF.toString(b, (cstring val) { se(val, f); return val.length; });
         else

--- a/src/ocean/text/convert/Formatter_test.d
+++ b/src/ocean/text/convert/Formatter_test.d
@@ -567,3 +567,12 @@ unittest
     // Used to work only with "{:X}", however this limitation was lifted
     assert(format("{X}", 42) == "2A");
 }
+
+unittest
+{
+    static void myFunc (inout char[] arg)
+    {
+        assert(format("{}", arg[0]) == "H");
+    }
+    myFunc("Hello World");
+}

